File: backend/src/main/kotlin/com/example/backend/dtos/CaisseDto.kt
package com.example.backend.dtos

import java.math.BigDecimal
import java.time.LocalDateTime

data class CaisseOuvertureRequestDto(
  val fondCaisseOuvert: BigDecimal,
  val ouvertureCaisse: String
)

data class CaisseDto(
  val id: Int?,
  val employeId: Int?,
  val employeNom: String?, // Pour affichage
  val dateOuvert: LocalDateTime?,
  val dateFerme: LocalDateTime?,
  val session: String?,
  val fondCaisseOuvert: BigDecimal?,
  val fondCaisseFerme: BigDecimal?,
  val etat: String?
)

data class CaisseClotureRequestDto(
  val fondCaisseFerme: Int,
  val fermetureCaisse: String
)

data class CaisseFermetureRequestDto(
  val fondCaisseFerme: BigDecimal,
  // Ajoutez d'autres champs si nécessaire pour la clôture, ex: totalVentesEspeces, totalDepenses
)

File: backend_py/pharmaPython/app/schemas/caisse_dto.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from decimal import Decimal


class CaisseOuvertureRequestDto(BaseModel):
  fondCaisseOuvert: float = Field(..., alias="fondCaisseOuvert")
  ouvertureCaisse: str = Field(..., alias="ouvertureCaisse")

  class Config:
    populate_by_name = True


class CaisseDto(BaseModel):
  id: Optional[int]
  employeId: Optional[int]
  nomEmploye: Optional[str]  # Pour affichage
  dateOuvert: Optional[datetime]
  dateFerme: Optional[datetime]
  session: Optional[str]
  fondCaisseOuvert: Optional[float]
  fondCaisseFerme: Optional[float]
  etat: Optional[str]


class CaisseClotureRequestDto(BaseModel):
  fondCaisseFerme: int
  fermetureCaisse: str


class CaisseFermetureRequestDto(BaseModel):
  fondCaisseFerme: float
  # Ajoutez d'autres champs si nécessaire pour la clôture
  # ex: totalVentesEspeces, totalDepenses
